django_emarsys/util.py

```diff
@@ -0,0 +1,97 @@
+
+log = logging.getLogger(__file__)
+
+
+def get_model(model_string):
+    chunks = model_string.split('.')
+    app_label = '.'.join(chunks[:-1])
+    model_name = chunks[-1]
+    return apps.get_model(app_label=app_label, model_name=model_name)
+
+
+def validate_configuration():
+    if not getattr(settings, 'EMARSYS_ACCOUNT', None):
+        log.warning("no EMARSYS_ACCOUNT setting set")
+
+    if not getattr(settings, 'EMARSYS_PASSWORD', None):
+        log.warning("no EMARSYS_PASSWORD setting set")
+
+    if not getattr(settings, 'EMARSYS_BASE_URI', None):
+        log.warning("no EMARSYS_BASE_URI setting set")
+
+    events = getattr(settings, 'EMARSYS_EVENTS', None)
+    if not events:
+        log.warning("no EMARSYS_EVENTS setting set")
+        return
+
+    VALID_ARGUMENT = re.compile(r'^[a-z][a-z_0-9]*$')
+    for event, data in events.items():
+        if not event:
+            log.warning("invalid event name: '{event}'".format(event=event))
+
+        names = set()
+        args = set()
+        for data in data:
+            if 'name' not in data:
+                log.error("missing data key for event '{event}': 'name'"
+                          .format(event=event))
+
+            else:
+                name = data['name']
+                if name in names:
+                    log.warning(
+                        "reused data name for event '{event}': '{name}'"
+                        .format(name=name, event=event))
+
+                if not name or not name.strip():
+                    log.warning(
+                        "invalid data name for event '{event}': '{name}'"
+                        .format(name=name, event=event))
+
+                names.add(name)
+
+            if 'arg' not in data:
+                log.error("missing data key for event '{event}': 'arg'"
+                          .format(event=event))
+            else:
+                arg = data['arg']
+                if arg in args:
+                    log.error(
+                        "reused data argument for event '{event}': '{arg}'"
+                        .format(arg=arg, event=event))
+
+                if not arg or not VALID_ARGUMENT.match(arg):
+                    log.error(
+                        "invalid data argument for event '{event}': '{arg}'"
+                        .format(arg=arg, event=event))
+
+                args.add(arg)
+
+            if 'model' not in data:
+                log.error("missing data key for event '{event}': 'model'"
+                          .format(event=event))
+            else:
+                model = data['model']
+                chunks = model.split('.')
+                if len(chunks) < 2:
+                    log.error("model for event '{event}' expected to be "
+                              "'<app_label>.<model_name>', got: '{model}'"
+                              .format(event=event, model=model))
+                else:
+                    try:
+                        get_model(model)
```

Comment:
`get_model` is a notorious bitch and can be used only after the models were loaded successfully. This limits the places where a call to this function can be placed. Just fyi.